include/RAJA/policy/cuda/teams.hpp

```diff
@@ -381,6 +381,137 @@ struct LoopExecute<cuda_global_thread<DIM0, DIM1, DIM2>, SEGMENT> {
   }
 };
 
+/*
+Reshape threads in a block into a 1D iteration space
+*/
+template<int ... dim>
+struct flatten_block_threads_direct{};
+
+using flatten_block_threads_xy_direct = flatten_block_threads_direct<0,1>;
+using flatten_block_threads_xz_direct = flatten_block_threads_direct<0,2>;
+using flatten_block_threads_yx_direct = flatten_block_threads_direct<1,0>;
+using flatten_block_threads_yz_direct = flatten_block_threads_direct<1,2>;
+using flatten_block_threads_zx_direct = flatten_block_threads_direct<2,0>;
+using flatten_block_threads_zy_direct = flatten_block_threads_direct<2,1>;
+
+using flatten_block_threads_xyz_direct = flatten_block_threads_direct<0,1,2>;
+using flatten_block_threads_xzy_direct = flatten_block_threads_direct<0,2,1>;
+using flatten_block_threads_yxz_direct = flatten_block_threads_direct<1,0,2>;
+using flatten_block_threads_yzx_direct = flatten_block_threads_direct<1,2,0>;
+using flatten_block_threads_zxy_direct = flatten_block_threads_direct<2,0,1>;
+using flatten_block_threads_zyx_direct = flatten_block_threads_direct<2,1,1>;
+
+template<int ... dim>
+struct flatten_block_threads_loop{};
+
+using flatten_block_threads_xy_loop = flatten_block_threads_loop<0,1>;
+using flatten_block_threads_xz_loop = flatten_block_threads_loop<0,2>;
+using flatten_block_threads_yx_loop = flatten_block_threads_loop<1,0>;
+using flatten_block_threads_yz_loop = flatten_block_threads_loop<1,2>;
+using flatten_block_threads_zx_loop = flatten_block_threads_loop<2,0>;
+using flatten_block_threads_zy_loop = flatten_block_threads_loop<2,1>;
+
+using flatten_block_threads_xyz_loop = flatten_block_threads_loop<0,1,2>;
+using flatten_block_threads_xzy_loop = flatten_block_threads_loop<0,2,1>;
+using flatten_block_threads_yxz_loop = flatten_block_threads_loop<1,0,2>;
+using flatten_block_threads_yzx_loop = flatten_block_threads_loop<1,2,0>;
+using flatten_block_threads_zxy_loop = flatten_block_threads_loop<2,0,1>;
+using flatten_block_threads_zyx_loop = flatten_block_threads_loop<2,1,1>;
```

Comment:
it sounds like you're going to combine these files in the future but this set of definitions is the same in both as far as I can tell and it could be better to only have it in one place